packages/firestore/src/local/memory_persistence.ts

```diff
@@ -119,3 +154,154 @@ export class MemoryPersistence implements Persistence {
 export class MemoryTransaction implements PersistenceTransaction {
   constructor(readonly currentSequenceNumber: ListenSequenceNumber) {}
 }
+
+export class MemoryLruDelegate implements ReferenceDelegate, LruDelegate {
+  private additionalReferences: ReferenceSet | null;
+  private orphanedSequenceNumbers: ObjectMap<
+    DocumentKey,
+    ListenSequenceNumber
+  > = new ObjectMap(k => encode(k.path));
+
+  readonly garbageCollector: LruGarbageCollector;
+
+  constructor(private readonly persistence: MemoryPersistence) {
+    this.garbageCollector = new LruGarbageCollector(this);
+  }
+
+  forEachTarget(
+    txn: PersistenceTransaction,
+    f: (q: QueryData) => void
+  ): PersistencePromise<void> {
+    return this.persistence.getQueryCache().forEachTarget(txn, f);
+  }
+
+  getTargetCount(txn: PersistenceTransaction): PersistencePromise<number> {
+    return this.persistence.getQueryCache().getTargetCount(txn);
+  }
+
+  forEachOrphanedDocumentSequenceNumber(
+    txn: PersistenceTransaction,
+    f: (sequenceNumber: ListenSequenceNumber) => void
+  ): PersistencePromise<void> {
+    this.orphanedSequenceNumbers.forEach((_, sequenceNumber) =>
+      f(sequenceNumber)
+    );
+    return PersistencePromise.resolve();
+  }
+
+  setInMemoryPins(inMemoryPins: ReferenceSet): void {
+    this.additionalReferences = inMemoryPins;
+  }
+
+  removeTargets(
+    txn: PersistenceTransaction,
+    upperBound: ListenSequenceNumber,
+    activeTargetIds: ActiveTargets
```

Comment:
The argument name should just be `activeTargets`.

Reply:
This has already been changed once, I'd prefer to leave it as is at this point.